pyccel/ast/basic.py

```diff
@@ -642,14 +598,14 @@ def static_class_type(cls):
         This function is static and will return an AttributeError if the
         class does not have a predetermined order.
         """
-        return cls._class_type # pylint: disable=no-member
+        return cls._static_type # pylint: disable=no-member
```

Comment:
@yguclu I wonder if we should disable the pylint error `no-member` project wide? I seem to find myself regularly disabling it when it is used in a superclass. In this case it is close to being a false-positive. In any case the coverage should ensure that the property does actually exist when it is used

Reply:
I think this approach would lead to a lot of code duplication. For example this would mean that almost every class inheriting from `TypedAstNode` would need to add:
```python
@property
def rank(self):
    return self._rank
```

Reply:
Possibly yes. This approach focuses on enforcing common interfaces, and it does not necessarily minimize code duplication. Anyway, for scalar types you could have an intermediate base class which always returns 0 and does not require `_rank`...

On the other hand, the problem with our corrent approach is that the developer must know (somehow) that they need to give the private attribute `_rank` to their class, but this may not be obvious just by looking at the base class. The ABC approach makes the requirement of the `rank` property explicit, with a docstring which clearly describes what `rank` is supposed to be, and does not care about the internal representation as a private attribute.

Much food for thought here, but beyond the scope of this PR.